### DISCUSSION TITLE
OpenAM installation and Entity Import/export

### DISCUSSION THREAD

**Discussion by capmatch** [OpenIdentityPlatform/OpenAM#474] [Q&A] [answered]
Hello,
i now designing a CI/CD to setup a openam environment from scratch. As i know after setup the tomcat and put the war. You may need to :
1. manually do the configuration store setup
2. Use the console to upload Entity (SP or IDP in saml2) 

From the automation point of view, is there any tool/steps may automate the 1 and 2 steps? I know for configuration store , there is a SSOconfigruationTools.zip which do similar thing (but seems not much documentation). For 2, is that the ssoAdminTools are made for that? But i find that it required manual installation. i also find that the ssoadmin.jsp may do similar thing but there is not API documentation. Any one may share on that?

thx

capho

**Comment by maximthomas** [comment, marked as answer]
@capmatch I think you can use Selenium. Also you can use https://github.com/OpenIdentityPlatform/OpenAM/blob/master/openam-server/src/test/java/org/openidentityplatform/openam/test/integration/IT_Setup.java as an example.
